src/library/clementine/clementineplaylistmodel.h

```diff
@@ -0,0 +1,38 @@
+#pragma once
+
+#include <QHash>
+#include <QtSql>
+
+#include "library/basesqltablemodel.h"
+#include "library/clementine/clementinedbconnection.h"
+#include "library/trackcollection.h"
+#include "library/trackmodel.h"
+
+class ClementinePlaylistModel : public BaseSqlTableModel {
+    Q_OBJECT
+  public:
+    ClementinePlaylistModel(QObject* pParent,
+            TrackCollectionManager* pTrackCollectionManager,
+            ClementineDbConnection* pConnection);
+    ~ClementinePlaylistModel() final;
+
+    void setTableModel(int playlistId);
+
+    TrackPointer getTrack(const QModelIndex& index) const final;
+    TrackId getTrackId(const QModelIndex& index) const final;
+
+    QString getTrackLocation(const QModelIndex& index) const final;
+    bool isColumnInternal(int column) final;
+
+    Qt::ItemFlags flags(const QModelIndex& index) const final;
+    Capabilities getCapabilities() const final;
+
+  private:
+    TrackId doGetTrackId(const TrackPointer& pTrack) const final;
+
+    QString getFieldString(const QModelIndex& index, const QString& fieldName) const;
+    QVariant getFieldVariant(const QModelIndex& index, const QString& fieldName) const;
+
+    ClementineDbConnection* m_pConnection;
```

Comment:
Don't use plain, unmanaged pointers with undefined lifetime.

Reply:
added this in https://github.com/mixxxdj/mixxx/pull/3173/commits/f2a59de119ecefa97cf8d9ae8821a79d37650a05

Reply:
Why did you mark this as resolved?? The lifetime of the pointer is still unmanaged. This should either be a unique_ptr or shared_ptr.

Reply:
ClementineDbConnection is not a parented QObject.